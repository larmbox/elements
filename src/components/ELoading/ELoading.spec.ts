import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { ELoading } from '../..';

test('mount component', async () => {
  expect(ELoading).toBeTruthy();

  const wrapper = mount(ELoading, {
    props: {
      id: 'id',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
  expect(wrapper.html()).toContain('ELoading');
});

test('register css variables', async () => {
  mount(ELoading);
  expect(
    document.getElementById('EThemeProvider--global')!.innerText
  ).toContain('ELoading');
});

test('inherit attributes', async () => {
  const component = mount(ELoading, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
