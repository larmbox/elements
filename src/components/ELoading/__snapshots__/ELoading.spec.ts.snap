// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div id=\\"id\\" class=\\"ELoading ELoading--spinner ELoading--md ELoading--primary\\" aria-busy=\\"true\\" role=\\"progressbar\\">
  <div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</div>"
`;
